qiskit/utils/quantum_instance.py

```diff
@@ -304,6 +305,23 @@ def execute(self,
         if not had_transpiled:
             circuits = self.transpile(circuits)
 
+        # fix for providers that don't support QasmQobj until a bigger refactor happens
+        from qiskit.providers import BackendV1
+        if isinstance(self._backend, BackendV1) and \
+                not is_aer_provider(self._backend) and \
+                not is_basicaer_provider(self._backend) and \
+                not is_ibmq_provider(self._backend):
```

Comment:
I'm not sure how I feel about this, qobj support is dated and will be going away in ~<3months for the ibmq provider; for aer and basic aer it'll be around a bit longer, but not much. They're also all emitting warnings either a `DeprecationWarning` or a `PendingDeprecationWarning` which means anytime someone runs an algorithm on up-to-date versions of these backends we'll be emitting a warning for each job. But, I get that this is a workaround just to fix the bug for now. Maybe just expand on the comment here that we can't rely on Qobj for much longer even if the backends are from ibmq, aer, or basicaer.

Reply:
I agree that the ideal would be to just not use Qobj anymore however many things will change possibly including how AQGD works and the whole error mitigation logic. Excluding error mitigation, currently, if I use Aer without Qobj, I get several unit tests failures while the same pass using BasicAer without QObj.

Reply:
I expect this is because when running against aer it expects to be able to hack the qobj up to pass a paramterized circuit, and the circuit interface to aer is missing equivalent functionality (it was hacked in aer as a custom path for aqua and not really exposed as real interface). We should open an issue on aer to add an alternative for passing in parameterized circuits and a list of bound values as a kwarg to run without needing qobj.

Reply:
> We should open an issue on aer to add an alternative for passing in parameterized circuits and a list of bound values as a kwarg to run without needing qobj.

Actually it would be nice for for such a capability no matter the backend.

Reply:
Well most backends don't have a method to natively run parameterized circuits over a parameter list like aer does, but in such cases the provider could just expand the circuits before sending them to the backend.

I think the place to define this would be in the next version of the backend interface (which I have a rfc/prototype up in #5885 )